ocaml/xapi/dbsync_master.ml

```diff
@@ -156,12 +156,20 @@ let release_locks ~__context =
       Xapi_vm_lifecycle.force_state_reset ~__context ~self ~value:`Halted
     )
     vms ;
-  (* All VMs should have their scheduled_to_be_resident_on field cleared *)
-  List.iter
-    (fun self ->
-      Db.VM.set_scheduled_to_be_resident_on ~__context ~self ~value:Ref.null
-    )
-    (Db.VM.get_all ~__context)
+  (* Clear all assignments that are only scheduled *)
```

Comment:
I understand these `scheduled_*` fields are used only by the coordinator to `remember` something within XAPI calls. So it will be safe to clean all of these fields when the xapi on coordinator is in starting up, as at the moment no any xapi calls are in progress.